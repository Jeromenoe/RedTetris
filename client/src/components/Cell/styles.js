import styled from 'styled-components'

export const StyledCell = styled.div`
	background: #${props => props.color}
`