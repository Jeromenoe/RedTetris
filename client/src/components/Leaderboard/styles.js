import styled from 'styled-components'

export const StyledLeaderboardContainer = styled.div`
	visibility: ${props => props.gameOver === true ? "visible" : "hidden"};
	opacity: ${props => props.gameOver === true ? "1" : "0"};
	transition:visibility 0.3s linear,opacity 0.3s linear;

	background-color: rgba(0, 0, 0, 0.7);
	position: absolute;
	left: 0; 
	top: 0; 
	width: 100%; 
	height:100%;
	display: flex;
	justify-content: center;
	align-items: center;
	/* font-family: 'Press Start 2P', cursive; */
`

export const StyledLeaderboard = styled.div`
	/* background-color: hsl(210, 70%, 45%); */
	background-color: rgb(200, 54, 67);
	box-shadow: rgba(0, 0, 0, 0.1) 0px 4px 12px;
	color: #fff;
	text-align: center;
	border-radius: 6px;
	padding: 8px 0;
	width: 50%;
	min-width: 310px;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
`