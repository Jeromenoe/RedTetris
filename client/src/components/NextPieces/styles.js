import styled from 'styled-components'

export const StyledBlockNextPieces = styled.div`
	background: rgb(229, 231, 233);
	margin: 10px 20px 11px 8px;
	padding: 5px 5px;
	border: 1px solid rgb(231, 76, 60);
	width: 100px;
	display: flex;
	flex-direction: column;
	align-items: center;
`

export const StyledContainerNextPiece = styled.div`
	height: 40px;
	padding: 5px 10px;
	display: flex;
	align-items: center;
`


export const StyledNextPiece = styled.div`
	display: grid;
	grid-template-rows: repeat(
		${props => props.height}, 19px);
	grid-template-columns: repeat(${props => props.width}, 19px);
	width: calc(${props => props.width} * 20px);
	grid-gap: 1px ;
`

export const StyledTitleNextPieces = styled.h1`
	border-bottom: 1px solid rgb(231, 76, 60);
	margin: 0px;
	margin-bottom: 4px;
	color: rgb(231, 76, 60);
	font-size: 12px;
	font-family: 'Press Start 2P', cursive;
	width: 100%;
`