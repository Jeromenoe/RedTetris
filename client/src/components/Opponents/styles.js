import styled from 'styled-components'

export const StyledOpponentSpectrum = styled.div`
	display: grid;
	grid-template-rows: repeat(
		${props => props.height}, 7px);
	grid-template-columns: repeat(${props => props.width}, 7px);
	border: 2px solid rgb(231, 76, 60);
	border-top: 0px;
	background-color: rgb(93, 173, 226);
	width: 70px;
	height: 160px;
	margin: 6px;
`

export const StyledOpponentsBlock = styled.div`
	display: grid;
	grid-row-gap: 15px;
	grid-template-rows: repeat(2, 170px);
	grid-template-columns: repeat(${props => props.nbOpponents === 1 ? 1 : (props.nbOpponents / 2)}, 80px);
	grid-auto-flow: column;
	margin-left: 20px;
`

export const StyledOpponentDisplay = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: center;
	align-items: center;
	color: rgb(240, 240, 240);
`