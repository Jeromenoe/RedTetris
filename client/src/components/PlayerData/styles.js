import styled from 'styled-components'

export const StyledPlayerData = styled.div`
	align-self: flex-end;
	text-align: right;
	color: rgb(240, 240, 240);
	font-family: 'Press Start 2P', cursive;
	font-size: 12px;
`