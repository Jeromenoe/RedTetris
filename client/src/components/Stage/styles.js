import styled from 'styled-components'

export const StyledStage = styled.div`
		display: grid;
		grid-template-rows: repeat(
			${props => props.height}, ${props => props.cellSize + 'px'});
		grid-template-columns: repeat(${props => props.width}, ${props => props.cellSize + 'px'});
		grid-gap: 1px ;
		border: 2px solid rgb(231, 76, 60) ;
		background: rgb( 189, 195, 199);
		/* background-color: #ecf0f1; */
		width: ${props => props.width * props.cellSize + props.width + 'px'};
		height: ${props => props.height * props.cellSize + props.height + 'px'};
		margin: 10px;
		padding: 1px;
		margin-right: 0px;
`