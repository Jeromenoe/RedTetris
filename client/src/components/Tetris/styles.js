import styled from 'styled-components'

export const StyledTetrisWrapper = styled.div`
	display: flex;
	flex-direction: column;
	align-items: center;
	/* justify-content: center; */
	height: 100%;
	width: 100%;
`

export const StyledTetris = styled.div`
	position: relative;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	border-radius: 8px;
	height: 65vh;
	width: 70vw;
	min-width: 750px;
	min-height: 550px;
	background-color: rgba(110, 110, 110, 0.3);
	border: 1px solid;
	box-shadow: rgba(0, 0, 0, 0.1) 0px 4px 12px;
`

export const StyledBoard = styled.div`
	display: flex;
	justify-content: center;
`

export const StyledTetrisContainer = styled.div`
	width: 100%;
	height: 100%;
	position: relative;
	display: flex;
	align-items: center;
	justify-content: space-evenly;
`