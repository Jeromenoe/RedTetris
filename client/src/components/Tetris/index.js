import React, { useContext } from 'react';

// Component
import Stage from '../Stage/index';
import { StyledTetris, StyledTetrisWrapper, StyledBoard, StyledTetrisContainer } from './styles';
import {NextPieces} from '../NextPieces/index';
import { PlayerData } from '../PlayerData/index'
import { Opponents } from '../Opponents/index';

import { SocketContext } from '../../context/socket';
// import store from '../../store/index';
import { useStore } from 'react-redux'
import { updateData } from '../../actions/game';
import { Leaderboard } from '../Leaderboard/index';


const Tetris = () => {
	// console.log('re-render Tetris');
	const socket = useContext(SocketContext);
	const store = useStore();
	const state = store.getState();

	const sendPlayerLeaveSignal = () => {
        socket.emit('playerLeave', null);
        let gameCloned = {
			name: null,
			isStarted: false,
			isOver: false,
			opponents: [],
			players: [],
			removeOwnIndestructibleLine: false,
			difficulty: 2,
			nbOfNextTetromino: 3,
        };
        store.dispatch(updateData({game: gameCloned}));
	};
	const move = (keyCode, isOver) => {
		var offsetX = 0;
		var offsetY = 0;
		var drop = 0;
		var rotation = 0;

		if(!isOver) {
			if (keyCode === 37) {
				offsetX = -1;
			} else if (keyCode === 39) {
				offsetX = 1;
			} else if (keyCode === 40) {
				offsetY = 1;
			} else if (keyCode === 38) {
				rotation = 1;
			} else if (keyCode === 32) {
				drop = 1
			}
		}
		return {offsetX, offsetY, drop, rotation};
	}
	const handleKey = ({ keyCode }, isOver) => {
		const { offsetX, offsetY, drop, rotation } = move(keyCode, isOver);
		if (offsetX || offsetY || drop || rotation) {
			socket.emit('USER_ACTION', {offsetX, offsetY, drop, rotation});
		}
	}
	return (
		<StyledTetrisWrapper id="styledTetrisWrapper" role="button" tabIndex="0" onKeyDown={e => handleKey(e, state.game.isOver)}>
				<h1 className="title" style={{marginTop: '60px', marginBottom: '40px'}}>RED TETRIS</h1>
				<StyledTetris>
					<div style={{'fontFamily': '\'Press Start 2P\', cursive', 'fontSize': '12px', 'margin': '5px 0', 'color': 'rgb(55, 55, 55)'}}>
						{store.getState().me.name}@{store.getState().game.name}
					</div>
					<div style={{width: '100%', height: '1px', background: 'rgba(0,0,0,0.7)'}}></div>
					<StyledTetrisContainer>
						<div style={{display: 'flex'}}>
							<PlayerData />
							<aside>
								<StyledBoard>
									<Stage />
									<aside>
										<NextPieces />
									</aside>
								</StyledBoard>
							</aside>
						</div>
						<Opponents />
						<button className="LeaveButton" id="leavePlayingGame" onClick={sendPlayerLeaveSignal}>LEAVE</button>
					</StyledTetrisContainer>
				</StyledTetris>
				<Leaderboard />
		</StyledTetrisWrapper>
	)
}

export default Tetris;