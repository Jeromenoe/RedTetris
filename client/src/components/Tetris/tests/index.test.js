import React from 'react';
import Tetris from '../index';
import renderer from 'react-test-renderer';
import configureStore from "redux-mock-store";
import {Provider} from 'react-redux';
import toJson from "enzyme-to-json";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { mount, configure } from 'enzyme';
import { StyledTetrisWrapper } from "../styles";

import { SocketContext, socket } from "../../../context/socket"


configure({adapter: new Adapter()});


describe('Testing Tetris Component', () => {
	test('Test for snapshot', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				name: null,
				isStarted: false,
				isOver: false,
				opponents: [],
				players: [],
				removeOwnIndestructibleLine: false,
				difficulty: 2,
				nbOfNextTetromino: 3,
			},
			me: {
				name: null,
				hasLost: false,
				stage: [[]],
				nextPieces: [],
				rows: 0,
				level: 0,
				score: 0
			}
		};
		const store = mockStore(initialState);
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><Tetris /></SocketContext.Provider></Provider>);
		expect(toJson(wrapper)).toMatchSnapshot();
	  });
	  test('Test for input', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				name: 'test',
				isStarted: false,
				isOver: false,
				opponents: [],
				players: [],
				removeOwnIndestructibleLine: false,
				difficulty: 2,
				nbOfNextTetromino: 3,
			},
			me: {
				name: null,
				hasLost: false,
				stage: [[]],
				nextPieces: [],
				rows: 0,
				level: 0,
				score: 0
			}
		};
		const store = mockStore(initialState);
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><Tetris /></SocketContext.Provider></Provider>);
		wrapper.find(StyledTetrisWrapper).simulate('keyDown', { keyCode: 37 });
		wrapper.find(StyledTetrisWrapper).simulate('keyDown', { keyCode: 39 });
		wrapper.find(StyledTetrisWrapper).simulate('keyDown', { keyCode: 40 });
		wrapper.find(StyledTetrisWrapper).simulate('keyDown', { keyCode: 38 });
		wrapper.find(StyledTetrisWrapper).simulate('keyDown', { keyCode: 32 });
		wrapper.find(StyledTetrisWrapper).simulate('keyDown', { keyCode: 30 });
		wrapper.find('#leavePlayingGame').simulate('click');
	  });

	  test('Test for input with wrong keycode', async () => {
		const mockStore = configureStore();
		const initialState = {
			game: {
				name: 'test',
				isStarted: false,
				isOver: true,
				opponents: [],
				players: [],
				removeOwnIndestructibleLine: false,
				difficulty: 2,
				nbOfNextTetromino: 3,
			},
			me: {
				name: null,
				hasLost: false,
				stage: [[]],
				nextPieces: [],
				rows: 0,
				level: 0,
				score: 0
			}
		};
		const store = mockStore(initialState);
		const wrapper = mount(<Provider store={store}><SocketContext.Provider value={socket}><Tetris /></SocketContext.Provider></Provider>);
		wrapper.find(StyledTetrisWrapper).simulate('keyDown', { keyCode: 37 });
	});
});