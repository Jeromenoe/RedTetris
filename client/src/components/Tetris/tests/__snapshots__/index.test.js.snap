// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Tetris Component Test for snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Tetris>
    <styled.div
      id="styledTetrisWrapper"
      onKeyDown={[Function]}
      role="button"
      tabIndex="0"
    >
      <div
        className="sc-dlnjwi iPWQhR"
        id="styledTetrisWrapper"
        onKeyDown={[Function]}
        role="button"
        tabIndex="0"
      >
        <h1
          className="title"
          style={
            Object {
              "marginBottom": "40px",
              "marginTop": "60px",
            }
          }
        >
          RED TETRIS
        </h1>
        <styled.div>
          <div
            className="sc-hKFxyN idlCam"
          >
            <div
              style={
                Object {
                  "color": "rgb(55, 55, 55)",
                  "fontFamily": "'Press Start 2P', cursive",
                  "fontSize": "12px",
                  "margin": "5px 0",
                }
              }
            >
              @
            </div>
            <div
              style={
                Object {
                  "background": "rgba(0,0,0,0.7)",
                  "height": "1px",
                  "width": "100%",
                }
              }
            />
            <styled.div>
              <div
                className="sc-jSFjdj djwQrH"
              >
                <div
                  style={
                    Object {
                      "display": "flex",
                    }
                  }
                >
                  <PlayerData>
                    <div
                      style={
                        Object {
                          "alignSelf": "flex-end",
                          "marginBottom": "6px",
                          "marginRight": "6px",
                          "width": "70px",
                        }
                      }
                    >
                      <styled.div>
                        <div
                          className="sc-jrsJWt ilCZvT"
                        >
                          <div
                            style={
                              Object {
                                "marginBottom": "15px",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "marginBottom": "6px",
                                }
                              }
                            >
                              Level
                            </div>
                            <span
                              style={
                                Object {
                                  "paddingRight": "6px",
                                }
                              }
                            >
                              0
                            </span>
                          </div>
                          <div
                            style={
                              Object {
                                "marginBottom": "15px",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "marginBottom": "6px",
                                }
                              }
                            >
                              Score
                            </div>
                            <span
                              style={
                                Object {
                                  "paddingRight": "6px",
                                }
                              }
                            >
                              0
                            </span>
                          </div>
                          <div
                            style={
                              Object {
                                "marginBottom": "20px",
                              }
                            }
                          >
                            <div
                              style={
                                Object {
                                  "marginBottom": "6px",
                                }
                              }
                            >
                              Lines
                            </div>
                            <span
                              style={
                                Object {
                                  "paddingRight": "6px",
                                }
                              }
                            >
                              0
                            </span>
                          </div>
                        </div>
                      </styled.div>
                    </div>
                  </PlayerData>
                  <aside>
                    <styled.div>
                      <div
                        className="sc-eCApnc kQHbck"
                      >
                        <Stage>
                          <styled.div
                            cellSize={21}
                            height={1}
                            width={0}
                          >
                            <div
                              className="sc-gtsrHT cSPJwE"
                              height={1}
                              width={0}
                            />
                          </styled.div>
                        </Stage>
                        <aside>
                          <NextPieces>
                            <div>
                              <styled.div>
                                <div
                                  className="sc-gKAaRy jMDiMd"
                                >
                                  <styled.h1>
                                    <h1
                                      className="sc-pNWdM chyiGP"
                                    >
                                      NEXT
                                    </h1>
                                  </styled.h1>
                                </div>
                              </styled.div>
                            </div>
                          </NextPieces>
                        </aside>
                      </div>
                    </styled.div>
                  </aside>
                </div>
                <Opponents />
                <button
                  className="LeaveButton"
                  id="leavePlayingGame"
                  onClick={[Function]}
                >
                  LEAVE
                </button>
              </div>
            </styled.div>
          </div>
        </styled.div>
        <Connect(SubLeaderbord)>
          <SubLeaderbord
            dispatch={[Function]}
            gameOver={false}
          >
            <styled.div
              gameOver={false}
            >
              <div
                className="sc-dIsUp uSCOS"
              >
                <styled.div>
                  <div
                    className="sc-bqGGPW eLAWwL"
                  >
                    <strong
                      style={
                        Object {
                          "alignSelf": "center",
                          "borderBottom": "1px solid #ddd",
                          "display": "block",
                          "padding": "5px",
                          "width": "90%",
                        }
                      }
                    >
                      Game Over
                    </strong>
                    <div
                      className="player-score-container"
                    >
                      <div
                        className="player-score-rank"
                      >
                        Rank
                      </div>
                      <div
                        className="player-score-name"
                      >
                        Name
                      </div>
                      <div
                        className="player-score-pts"
                      >
                        Score
                      </div>
                      <div
                        className="player-score-lines"
                      >
                        Lines
                      </div>
                      <div
                        className="player-score-lvl"
                      >
                        Level
                      </div>
                    </div>
                    <div
                      className="btn-leaderboard-container"
                    >
                      <button
                        className="btn-leave-leaderboard"
                        onClick={[Function]}
                      >
                        Waiting Room
                      </button>
                      <button
                        className="btn-leave-leaderboard"
                        id="btn-leave-in-game"
                        onClick={[Function]}
                      >
                        Leave
                      </button>
                    </div>
                  </div>
                </styled.div>
              </div>
            </styled.div>
          </SubLeaderbord>
        </Connect(SubLeaderbord)>
      </div>
    </styled.div>
  </Tetris>
</Provider>
`;
